src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -334,6 +337,11 @@ CompleteShardPlacementTransactions(XactEvent event, void *arg)
 	shardConnectionHash = NULL;
 	XactModificationLevel = XACT_MODIFICATION_NONE;
 	subXactAbortAttempted = false;
+
+	if (MultiShardCommitProtocol == COMMIT_PROTOCOL_BARE)
+	{
+		MultiShardCommitProtocol = SavedMultiShardCommitProtocol;
+	}
```

Comment:
The `whateva` in your example needs to be a string representing one of the GUC values, right?

I spent some time looking into this, but thought I would necessarily have to add my special _bare_ value to the list of known `GUC` values; meaning it would be visible to the end-user and be settable by them. To avoid that, I'd probably have to write some GUC hook, right? The "just add another variable" hack seemed better?